File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int book[3];

void book_count(string text);

int main(void)
{
    string text = get_string("Text : ");

    book_count(text);

    float let = book[0];
    float wor = book[1];
    float set = book[2];

    float l = let/wor * 100.00;
    float s =  set/wor * 100.00;
    int grade = round(0.0588 * l - 0.296 * s - 15.8);

    if (grade > 16)
    {
    printf("Grade 16+\n");
    }
    else if(grade < 1)
    {
    printf("Before Grade 1\n");
    }
     else
     {
    printf("Grade %i\n", grade);
     }

}

void book_count(string text)
{
    int le = 0;
    int wo = 0;
    int se = 0;
    for (int i = 0 , n = strlen(text) ; i < n ; i++)
    {
       int val = text[i];

       if(val >64 && val <91 )
       {
            le++;
       }
       else if(val >96 && val <123 )
       {
            le++;
       }
        else if(val  == 32 )
       {
            wo++;
       }
        else if(val  == 46 ||  val  == 33 || val  == 63 )
       {
            se++;
       }
       else
       {
       }
    }
    wo++;
    book[0] = le;
    book[1] = wo;
    book[2] = se;
}

File: sentimental-readability/readability.py
from cs50 import get_string
import sys

def main():
    try:
        text = get_string("text : ")
        le = count_letters(text)
        #print(f"total characters {le}")
        wo = count_words(text)
        #print(f"total words {wo}")
        se = count_sentenses(text)
        #print(f"total sentenses {se}")
        l = le/wo * 100
        s = se/wo * 100
        index = round(0.0588 * l - 0.296 * s - 15.8)
        #print(f"score: {index}")
        if (index < 1):
            print("Before Grade 1")
        elif(index > 15):
            print("Grade 16+")
        else:
            print(f"Grade {index}")
    except ValueError:
        sys.exit(1)


def count_letters(text):
    valid_chars =['a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z']
    ttl_le = 0
    for l in text :
        if l.lower() in valid_chars :
            ttl_le += 1
    return ttl_le


def count_words(text):
    valid_chars =[' ']
    ttl_wo = 1
    for l in text :
        if l.lower() in valid_chars :
            ttl_wo += 1
    return ttl_wo

def count_sentenses(text):
    valid_chars =['.','!','?']
    ttl_se = 0
    for l in text :
        if l.lower() in valid_chars :
            ttl_se += 1
    return ttl_se


main()
